### ISSUE TITLE
[valid-expect-in-promise] Another false positive

### ISSUE THREAD

**Issue by srmagura** [jest-community/eslint-plugin-jest#947]
Hello, it's me again 😆. Thank you for the extremely quick fix to #930.

There are other cases where the rule is erroring even though the promise is (indirectly) awaited:

```js
it("throws then resolves", async () => {
  const p = Promise.reject(new Error("myError")).then(undefined, (e) => {
    expect(e).toBeInstanceOf(Error);

    return "handled";
  });

  await expect(p).resolves.toBe("handled");
});

it("throws twice", async () => {
  const p = Promise.reject(new Error("myError")).then(undefined, (e) => {
    expect(e).toBeInstanceOf(Error);

    throw new Error("myError2");
  });

  await expect(p).rejects.toThrow("myError2");
});
```

```text
  2:9  error  This promise should either be returned or awaited to ensure the expects in it's chain are called  jest/valid-expect-in-promise
  12:9  error  This promise should either be returned or awaited to ensure the expects in it's chain are called  jest/valid-expect-in-promise
```

**[Repository here](https://github.com/srmagura/eslint-plugin-jest-promise-repro2)**

### Versions

- "eslint": "^7.32.0",
- "eslint-plugin-jest": "^25.0.5"
- Node 14
